spec/function.dd

```diff
@@ -1242,9 +1242,18 @@ int foo(in int x, out int y, ref int z, int q);
     for its type)
 
     $(TROW $(D ref),   parameter is passed by reference)
-    $(TROW $(D scope), references in the parameter
-    cannot be escaped (e.g. assigned to a global variable).
-    Ignored for parameters with no references)
+    $(TROW $(D scope), $(ARGS
+    The parameter does not escape the function call
+    (e.g. by being assigned to a global variable).
+    Ignored for any parameter that is not of reference type.
+    Escape analysis is done for `@safe` functions, for other functions `scope` semantics must
+    be manually enforced.
+    As the parameter does not escape, the compiler can avoid heap-allocation of a unique argument
+    to a `scope` parameter. Due to this, such a function call can become `@nogc`.
```

Comment:
Yes, please see 5843070 for the changed wording.

Reply:
That's definitly better but I would change the example because passing `new T` to a scope parameter isn't `@nogc` (yet), You could also mention array literals (which were implemented recently)

Reply:
OK, I've now listed several things including array literals that can potentially avoid allocation.